packages/recordio/src/read.js

```diff
@@ -0,0 +1,25 @@
+export default function read(input) {
+  const records = [];
+
+  while (/^\d+\n.+/.test(input)) {
+    const delimiterPosition = input.indexOf("\n");
+    const recordPosition = delimiterPosition + 1;
+
+    const recordLength = parseInt(input.substring(0, delimiterPosition), 10);
+
+    if (recordLength > 0 && input.length >= recordPosition + recordLength) {
+      // Cut of the record length
+      input = input.substring(recordPosition, input.length);
```

Comment:
Don't reassign parameters even if they'er primitives and as such immutable, as this makes it harder to follow and maintain the code.